Removes support for unused optimize.* settings from core deprecations (#113296)
elastic/kibana: docs/migration/migrate_8_0.asciidoc

*Impact:* Any workflow that involved manually clearing generated bundles will have to be updated with the new path.

[float]
=== Legacy `optimize.*` settings are no longer supported

> *Details:*  The legacy optimizer has been removed and deprecations for unused `optimize.*` settings have been logged since 7.10. The following settings have been removed.

Reviewer: ```suggestion
*Details:*  The legacy optimizer has been removed and any `optimize.*` settings have been deprecated since 7.10. These settings have been removed as they are no longer in use.
```

Author: @lukeelmers Thanks for the suggestions!
Would you mind resubmitting your review please? The PR was still in draft but is open for review now 🙏

> *Impact:* Any of the following legacy `optimize.*` settings will prevent Kibana from starting up. Going forward, to run the `@kbn/optimizer` separately in development, pass `--no-optimizer` to `yarn start`. See https://github.com/elastic/kibana/pull/73154 for more details.

Reviewer: ```suggestion
*Impact:* Any of the legacy `optimize.*` settings will prevent Kibana from starting up. Going forward, to run the `@kbn/optimizer` separately in development, pass `--no-optimizer` to `yarn start`. See https://github.com/elastic/kibana/pull/73154 for more details.
```
